restcomm/restcomm.interpreter/src/main/java/org/restcomm/connect/interpreter/VoiceInterpreter.java

```diff
@@ -1761,6 +1764,9 @@ public void execute(final Object message) throws Exception {
 
                 // Start dialing.
                 call.tell(new Dial(), source);
+                // Set the timeout period.
+                final UntypedActorContext context = getContext();
+                context.setReceiveTimeout(Duration.create(timeout, TimeUnit.SECONDS));
```

Comment:
@gvagenas  told me about this timer being reset in a message is received. But we may need to keep the timer even if some message is received... I found this in Akka doc https://doc.akka.io/japi/akka/current/akka/actor/NotInfluenceReceiveTimeout.html to mark messages as whether they should reset receiveTimeout or not.

Honestly i think we should not use receiveTimeout for any kind of FSM timer, as most of the times is not really related to FSM/Actor inactivity, but other kind of logic (whether we receive the proper events in proper order,and FSM transitions to diff states accordingly). Maybe we need to use another timer infra to send timeout events to the actors.  I notice this more general purpose scheduler in akka https://doc.akka.io/docs/akka/2.5/scheduler.html, not sure is available in our akka version